### PR TITLE
Make API controllers log to BigQuery

### PR DESCRIPTION
**Pull request by duncanjbrown**
## Context

We missed these when we added `EmitRequestEvents` to `ApplicationController`... because the API controllers don't inherit from `ApplicationController`!

## Changes proposed in this pull request

- Create a new `ApplicationAPIController` and have the base API controllers inherit from it
- Deduplicate the BigQuery behaviour (and a couple of other common pieces) into it
- Forbid the creation of new subclasses of `ActionController::API` using a cop

## Guidance to review

Is the deduping correct?

## Link to Trello card

Bug spotted in Slack https://ukgovernmentdfe.slack.com/archives/CN1MCQCHZ/p1643289785245700

## Things to check

- [X] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [X] This code does not rely on migrations in the same Pull Request
- [X] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [X] API release notes have been updated if necessary
- [X] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: app/controllers/register_api/applications_controller.rb [hunk 1]
```diff
@@ -1,7 +1,6 @@
 module RegisterAPI
-  class ApplicationsController < ActionController::API
+  class ApplicationsController < ApplicationAPIController
     include ServiceAPIUserAuthentication
-    include RemoveBrowserOnlyHeaders
 
     rescue_from ActionController::ParameterMissing, with: :parameter_missing
```
[the remark this answers is not in the document]
Comment (duncanjbrown) [reply app/controllers/register_api/applications_controller.rb:5]: 🙏

**Review by cwrw — APPROVED** [review]
💯

**Comment by duncanjbrown** [comment]
Thanks both! Just straightened out the Rubocop requires, which were confusing `erblint`, and will merge

File: app/controllers/register_api/applications_controller.rb [hunk 2]
```diff
@@ -1,9 +1,7 @@
 module RegisterAPI
-  class ApplicationsController < ActionController::API
+  class ApplicationsController < ApplicationAPIController
     include ServiceAPIUserAuthentication
-    include RemoveBrowserOnlyHeaders
 
-    rescue_from ActionController::ParameterMissing, with: :parameter_missing
     rescue_from ParameterInvalid, with: :parameter_invalid
```
Comment (despo) [inline app/controllers/register_api/applications_controller.rb:5]: ✂️

**Pull request merged**